include/mlir/Dialect/StandardOps/Ops.td

```diff
@@ -881,6 +881,14 @@ def ShlISOp : IntArithmeticOp<"shlis"> {
   let summary = "signed integer shift left";
 }
 
+def ShrISOp : IntArithmeticOp<"shris"> {
+  let summary = "signed integer shift right";
+}
+
+def ShrIUOp : IntArithmeticOp<"shriu"> {
+  let summary = "unsigned integer shift right";
+}
```

Comment:
We need to specify better the behavior of these operation for edge cases (wrapping, etc.). In this dialect we don't have a notion of poison just like LLVM, and this dialect is intended to be re-used in multiple context, which makes it harder to narrow down the semantics too much (so I am not sure what is the best thing to do here)

Reply:
Hm, good question. Should we constrain the shift amount to be less than the bit-width for example? Unfortunately we cannot verify this so we would need to enforce that via additional instructions.
Should we discuss this on the mailing list?

Reply:
This is always welcome to be discussed on the mailing-list of course!